tests/operator/suite.go

```diff
@@ -126,6 +127,72 @@ func LoadChecks() {
 			testOperatorCatalogSourceBundleCount(c, &env)
 			return nil
 		}))
+
+	checksGroup.Add(checksdb.NewCheck(identifiers.GetTestIDAndLabels(identifiers.TestSingleNamespacedOperatorInstallationInTenantNamespace)).
+		WithSkipCheckFn(testhelper.GetNoOperatorsSkipFn(&env)).
+		WithCheckFn(func(c *checksdb.Check) error {
+			testOperatorInstallationInTenantNamespace(c, &env)
+			return nil
+		}))
+}
+
+// This function checks if SingleNamespaced Operators should only be installed in the tenant dedicated operator namespace
+func testOperatorInstallationInTenantNamespace(check *checksdb.Check, env *provider.TestEnvironment) {
+	check.LogInfo("Starting testInstalledSingleNamespaceOperatorInTenantNamespace")
+
+	var compliantObjects []*testhelper.ReportObject
+	var nonCompliantObjects []*testhelper.ReportObject
+
+	operatorNamespaces := make(map[string]bool)
+	for _, operator := range env.Operators {
+		operatorNamespaces[operator.Namespace] = true
+	}
+
+	for operatorNamespace := range operatorNamespaces {
+		check.LogInfo("Checking if namespace %s is an operator namespace", operatorNamespace)
+		doesNamespaceContainsOnlyOperatorPods, singleNamespacedCsvs, allNamespacedCsvs, podsBelongingToNoOperators, err := checkIfNamespaceContainsOnlyOperatorPods(operatorNamespace)
+
+		if err != nil {
+			check.LogError("Skipped - cannot proceed with operator namespace %s", operatorNamespace)
+			continue
+		}
+
+		var installedSingleNamespacedOperators string
+		if len(singleNamespacedCsvs) != 0 {
+			installedSingleNamespacedOperators = fmt.Sprintf("(%s)", strings.Join(singleNamespacedCsvs, ", "))
+		}
+
+		var msg string
+		if doesNamespaceContainsOnlyOperatorPods {
+			msg = fmt.Sprintf("Namespace is dedicated to single namespace operators %s ", installedSingleNamespacedOperators)
+			check.LogInfo(msg)
+			compliantObjects = append(compliantObjects, testhelper.NewNamespacedReportObject(msg, testhelper.Namespace, true, operatorNamespace))
+		} else {
+			const prefix = "Namespace with operators"
+			var msg string
+			if installedSingleNamespacedOperators != "" {
+				msg = fmt.Sprintf("%s (%s)", prefix, installedSingleNamespacedOperators)
+			} else {
+				msg = prefix
+			}
+
+			var allNamespacedOperators string
+			var nonCompliantMsg string
+			if len(allNamespacedCsvs) != 0 {
+				allNamespacedOperators = strings.Join(allNamespacedCsvs, ", ")
+				nonCompliantMsg = fmt.Sprintf("%s contains all-namespaced operators (%s) ", msg, allNamespacedOperators)
+			} else {
+				nonCompliantMsg = fmt.Sprintf("%s contains cluster-wide operators", msg)
```

Comment:
Is this message string ok? The all-namespaced (cluster-wide) operators were already checked at line 181.